guppylang/src/guppylang/decorator.py

```diff
@@ -639,8 +646,10 @@ def _with_optional_kwargs(
 
 
 @hide_trace
-def _parse_kwargs(kwargs: GuppyKwargs) -> UnitaryFlags:
-    """Parses the kwargs dict specified in the `@guppy` decorator."""
+def _parse_kwargs(kwargs: GuppyKwargs) -> tuple[UnitaryFlags, GuppyMetadata]:
```

Comment:
not very satisfying to return a pair here, should "UnitaryFlags" be conceptually expanded to "DecoratorHints" or something similar?